interview-questions.md

```diff
@@ -6,38 +6,69 @@ First, without external resources. Challenge yourself to answer from memory.
+$ git commit -m "initial commit"
+$ git push origin main
+$ bundle add rspec-rails
+$ rails generate rspec:install
+$ rails s
+$ git checkout -b animal-crud-actions
+$ rails generate resource Animal common_name:string scientific_binomial:string
+$ rails routes -E     routes cheat sheet
+$ rails db:migrate
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+Researched answer: Rails generator commands include controller, model and RSpec. This enables us to set up the folders and files needed so that we have the exact functionality we are looking for, but with elements like naming conventions set up appropriately. We can also create a model with columns & data types, a controller, RESTful routes and a view folder by using the generate resource command. 
 
-action: "GET" location: /students
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform? *Name and Purpose*
 
-action: "POST" location: /students
+action: "GET" location: /students The HTTP verb GET used on the path /students would perform the action of showing the index of students, which displays a list of all of the data available in students.
 
-action: "GET" location: /students/new
+action: "POST" location: /students The HTTP verb POST used on the path /students would perform the action of creating a new data set, so assuming that /students contains a list of names of students, it would create a new student name. 
 
-action: "GET" location: /students/2
+action: "GET" location: /students/new The HTTP verb GET used on the path /students/new would return an HTML form for creating a new student name (again, assuming that /students contains a list of names of students)
 
-action: "GET" location: /students/2/edit
+action: "GET" location: /students/2 The HTTP verb GET used on the path /students/2 would be SHOW, which would allow you to see the student at the index of 2. 
 
-action: "PATCH" location: /students/2
+action: "GET" location: /students/2/edit The HTTP verb GET is used on the path /students/2/edit would perform the action of allowing us to EDIT the information of the student at the index of 2.
 
-action: "DELETE" location: /students/2
+action: "PATCH" location: /students/2 The HTTP verb PATCH is used on the path /students/2 to perform the action of updating or editing a piece of data pertaining to the student at the index of 2.
+
+action: "DELETE" location: /students/2 The HTTP verb DELETE is used on the path /students/2 to perform the action of destroying the data found of the student at the index of 2.
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```

Comment:
these look good. I suggest adding what the user would see to complete those actions.